app/controllers/organized/messages_controller.rb

```diff
@@ -38,9 +38,10 @@ def organization_messages
     end
 
     def ensure_permitted_message_kind!
-      @kind = params[:organization_message].present? ? params[:organization_message][:kind_slug] : params[:kind]
-      @kind ||= current_organization.message_kinds
-      return true if current_organization.paid_for?(@kind)
+      kind = params[:organization_message].present? ? params[:organization_message][:kind_slug] : params[:kind]
+      @kinds = Array(kind).flatten if kind.present?
```

Comment:
`Array()` has nice semantics -- if `kind` isn't present here it'll return an empty array:

```rb
params = {}
Array(params.dig(:organization_message, :kind_slug) || params[:kind])
# []

params = {organization_message: {kind_slug: "slug"} }
Array(params.dig(:organization_message, :kind_slug) || params[:kind])
# [ "slug"]

params = {kind: "kind"}
Array(params.dig(:organization_message, :kind_slug) || params[:kind])
# [ "kind" ]

params = {organization_message: {kind_slug: ["slug", "slug2"]} }
Array(params.dig(:organization_message, :kind_slug) || params[:kind])
# [ "slug", "slug2" ]
```